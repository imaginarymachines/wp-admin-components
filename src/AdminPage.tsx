import React, { FC } from 'react';
const AdminPage: FC<{ children: any }> = ({ children }) => {
  return (
    <div className="wp-admin wp-core-ui js index-php auto-fold admin-bar customize-support svg sticky-menu">
      <div id="wpcontent">
        <div id="wpbody" role="main">
          <div id="wpbody-content">{children}</div>
        </div>
      </div>
    </div>
  );
};
export default AdminPage;
