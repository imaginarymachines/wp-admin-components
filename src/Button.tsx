import React, { FC, HTMLAttributes, ReactChild } from 'react';

export interface ButtonProps extends HTMLAttributes<HTMLDivElement> {
  /** custom content, defaults to 'the snozzberries taste like snozzberries' */
  children?: ReactChild;
  variant?: 'primary' | 'secondary';
  a?:"string";
  onClick?: () => void;
  isSmall?:boolean;
  isHero?:boolean;
}

const Button: FC<ButtonProps> = ({ children,variant,a,onClick,isSmall,isHero }) => {
   let className = `button  button-${variant ? variant : 'primary'}`;
   if( isSmall ){
     className = `${className} button-small`;
   }
   if( isHero ){
    className = `${className} button-hero`;
  }
   if( a ){
    return <a href={a} className={className}>{children}</a>;

   }
   return <button onClick={onClick} className={className}>{children}</button>
};
export default Button;
