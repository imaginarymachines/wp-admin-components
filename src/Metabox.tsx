import React, { FC,ReactChild } from 'react';

export interface MetaboxProps { children: ReactChild,title:string };
const Metabox: FC<MetaboxProps> = ({ children,title }) => {
  return (

    <div id="dashboard-widgets" className="metabox-holder">
    <div id="postbox-container-1" className="postbox-container">
        <div id="normal-sortables" className="meta-box-sortables ui-sortable">
            <div id="metabox" className="postbox">
                <button type="button" className="handlediv" aria-expanded="true">
                    <span className="screen-reader-text">Metabox collapse</span>
                    <span className="toggle-indicator" aria-hidden="true"></span>
                </button>
                <h2 className="hndle ui-sortable-handle">
                    <span>
                        {title}
                    </span>
                </h2>
                <div className="inside">
                    <div className="main">
                        {children}
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>

  );
};
export default Metabox;
