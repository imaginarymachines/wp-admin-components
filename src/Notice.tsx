import React, {FC, useMemo}from'react';

export type NoticeProps = {
    heading?:string;
    link?:string;
    description:string;
    type?:'success'|'error'|'info'|'warning',
    onDismiss?:() => void;
}
export const Notice: FC<NoticeProps>= ({ heading, link, description, type,onDismiss }) => {
    const className = useMemo( () => {
        let cN = `notice ${type ?`notice-${type}`:''}`;
        if( onDismiss){
            cN = `${cN} is-dismissible`;
        }
        return cN;
    },[type,onDismiss]);
    return(
    <div className={className}>
      {heading? <h2>{heading}</h2>:null}
      <p className="description">
          {link ? (
             <a href={link} rel="noopener" target="_blank">
               {description}
             </a>
          ): (
            <>{description}   </>
          )
        }
      </p>
    </div>
  );
}
