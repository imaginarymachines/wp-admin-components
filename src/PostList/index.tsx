import React, {FC,ReactChild} from 'react';


export type PostListItemProps = {children:ReactChild};
export const PostListItem : FC<PostListItemProps> = ({children}) => {

    return (<>{children}</>);
}

export type PostListProps = {children:ReactChild};

export const PostList : FC<{children:PostListProps}> = ({children}) => {
    return (
        <table className="wp-list-table widefat fixed striped posts">
            <thead>
                <tr>
                    <td id="cb" className="manage-column column-cb check-column">
                        <label className="screen-reader-text" htmlFor="cb-select-all-1">Select All</label>
                        <input id="cb-select-all-1" type="checkbox" />
                    </td>
                    <th scope="col" id="title" className="manage-column column-title column-primary sortable desc">
                        <span>Title</span>
                    </th>
                    <th scope="col" id="author" className="manage-column column-author">Author</th>
                    <th scope="col" id="categories" className="manage-column column-categories">Categories</th>
                    <th scope="col" id="tags" className="manage-column column-tags">Tags</th>
                    <th scope="col" id="comments" className="manage-column column-comments num sortable desc">
                        <span>
                            <span className="vers comment-grey-bubble" title="Comments">
                                <span className="screen-reader-text">Comments</span>
                            </span>
                        </span>
                    </th>
                    <th scope="col" id="date" className="manage-column column-date sortable asc">
                        <a href="#">
                            <span>Date</span><span className="sorting-indicator"></span>
                        </a>
                    </th>
                </tr>
            </thead>
            <tbody id="the-list">
                {children}
            </tbody>
        </table>
    )
}
