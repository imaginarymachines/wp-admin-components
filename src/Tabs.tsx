import React, {
  FC,
  HTMLAttributes,
  ReactChild,
  useMemo,
  useState,
} from 'react';

export interface TabProps {
  id: string;
  children: ReactChild;
  label: string;
}

export const Tab: FC<TabProps> = ({ id, children }) => {
  return <div id={id}>{children}</div>;
};

export interface TabButtonProps {
  id: string;
  label: ReactChild;
  isSelected: boolean;
  onClick: (id: string) => void;
}

export const TabButton: FC<TabButtonProps> = ({
  id,
  label,
  isSelected,
  onClick,
}) => {
  return (
    <li
      id={`tab-button-${id}`}
      aria-controls={`tab-${id}`}
      aria-selected={isSelected}
    >
      <a
        className="nav-tab"
        href={`#${id}`}
        onClick={(e) => {
          e.preventDefault();
          onClick(id);
        }}
      >
        {label}
      </a>
    </li>
  );
};

export interface TabContentProps extends HTMLAttributes<HTMLDivElement> {
  id: string;
  children: ReactChild;
  isSelected: boolean;
}

export const TabContent: FC<TabContentProps> = ({
  id,
  children,
  isSelected,
}) => {
  return (
    <div
      id={`tab-${id}`}
      className="tabs-content"
      role="tabpanel"
      aria-labelledby={`tab-button-${id}`}
      aria-hidden={!isSelected}
      style={
        isSelected
          ? {
              display: 'block',
              visibility: 'visible',
            }
          : {
              display: 'none',
              visibility: 'hidden',
            }
      }
    >
      {children}
    </div>
  );
};

export interface TabsProps extends HTMLAttributes<HTMLDivElement> {
  id: string;
  tabs: TabProps[];
  initialTab?: string;
}

export const Tabs: FC<TabsProps> = (props) => {
  const tabs = useMemo(() => {
    return props.tabs.length ? props.tabs : [];
  }, [props.tabs]);
  const [selectedTab, setSelectedTab] = useState<string | null>(() => {
    if (props.initialTab) {
      return props.initialTab;
    }
    if (!tabs.length) {
      return null;
    }
    return tabs[0].id;
  });

  const isSelectedTab = (tab: TabProps) => selectedTab === tab.id;

  return (
    <div id={props.id}>
      <h2 className="nav-tab-wrapper">
        <ul role="tablist">
          {tabs.map((tab: TabProps) => (
            <TabButton
              key={tab.id}
              isSelected={isSelectedTab(tab)}
              id={tab.id}
              label={tab.label}
              onClick={setSelectedTab}
            />
          ))}
        </ul>
      </h2>

      <>
        {tabs.map((tab: TabProps) => (
          <TabContent key={tab.id} isSelected={isSelectedTab(tab)} id={tab.id}>
            {tab.children}
          </TabContent>
        ))}
      </>
    </div>
  );
};
