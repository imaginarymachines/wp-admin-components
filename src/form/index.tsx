
export {Form,FormTable} from './FormTable'
export {FieldTr} from './FieldTr'
export {TrInput,TrSelect,TrSubmitButton} from './FormTableFields';
