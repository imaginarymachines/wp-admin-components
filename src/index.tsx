import Button, { ButtonProps } from './Button';
import { Tabs,TabsProps, TabProps } from './Tabs';
import {FormTable,Form,FieldTr,TrInput,TrSelect,TrSubmitButton} from './form'
export {Button, ButtonProps,Tabs,TabsProps, TabProps,FormTable,Form,FieldTr,TrInput,TrSelect,TrSubmitButton};
