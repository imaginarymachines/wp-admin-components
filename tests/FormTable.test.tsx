import {FormTable,Form, TrInput, TrSelect, TrSubmitButton} from '../src';
import React from 'react';
import snapshot from './snapshot';

let inputChange = (value:string) => {console.log(value)}
let inputProps = {
    id:'i-1',
    label:'Input One',
    onChange:inputChange,
    name: 'i-1',
    value: ''
}

let selectChange = (value:string) => {console.log(value)}
let selectProps = {
    id:'s-1',
    label:'Select',
    onChange:selectChange,
    name: 's-1',
    value: '',
    options:[
        {value: 'one', label:'One'},
        {value: 'two', label:'Two'}

    ]
}

let submitProps = {
    id:'sub-1',
    name: 'sub-1',
    value: 'Click Me'
}

describe('FormTable', () => {
    it( 'creates form', () => {
        const onSubmit = ()=>{};
        snapshot(<FormTable>
            <Form id="form-1" onSubmit={onSubmit}>
                <>
                    <TrInput
                        {...inputProps}
                    />
                    <TrSelect
                        {...selectProps}
                    />
                    <TrSubmitButton
                        {...submitProps}
                    />
                </>
            </Form>
        </FormTable>)
    });
})
