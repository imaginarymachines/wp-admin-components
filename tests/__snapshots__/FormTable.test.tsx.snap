// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTable creates form 1`] = `
<div>
  <table
    class="form-table"
    role="presentation"
  >
    <tbody>
      <form
        id="form-1"
      >
        <tr>
          <th
            scope="row"
          >
            <label
              for="i-1"
            >
              Input One
            </label>
          </th>
          <td>
            <input
              class="regular-text ltr"
              id="i-1"
              name="i-1"
              type="text"
              value=""
            />
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <label
              for="s-1"
            >
              Select
            </label>
          </th>
          <td>
            <select>
              <option
                class="level-0"
                value="one"
              >
                One
              </option>
              <option
                class="level-0"
                value="two"
              >
                Two
              </option>
            </select>
          </td>
        </tr>
        <p
          class="submit"
        >
          <input
            class="button button-secondary button-hero"
            id="sub-1"
            name="sub-1"
            type="submit"
            value="Click Me"
          />
        </p>
      </form>
    </tbody>
  </table>
</div>
`;
